save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/OrganizationController.kt

```diff
@@ -363,11 +372,34 @@ internal class OrganizationController(
         .switchIfEmptyToResponseException(HttpStatus.FORBIDDEN) {
             "Not enough permission for managing organization git credentials."
         }
-        .map {
-            gitService.delete(it, url)
-            ResponseEntity.ok("Git credential deleted")
+        .map { organization ->
+            // Find and remove all corresponding data to the current git repository from DB and file system storage
+            val git = gitService.getByOrganizationAndUrl(organization, url)
+            val testSuitesSources = testSuitesSourceService.findByGit(git)
+            testSuitesSources.forEach { testSuitesSource ->
+                val testSuites = testSuitesService.getBySource(testSuitesSource)
+                // Remove data from storage,
+                // since it is common for all test suites from one test suite source, it's enough to take any one of them
+                testSuites.firstOrNull()?.let {
+                    cleanupStorageData(it)
+                }
+                testSuitesService.deleteTestSuiteDto(testSuites.map { it.toDto() })
+                testSuitesSourceService.delete(testSuitesSource)
+            }
+            gitService.delete(organization, url)
+            ResponseEntity.ok("Git credentials and corresponding data successfully deleted")
         }
 
+    private fun cleanupStorageData(testSuite: TestSuite) {
+        testSuitesSourceSnapshotStorage.findKey(
+            testSuite.source.organization.name,
+            testSuite.source.name,
+            testSuite.version,
+        ).flatMap { key ->
+            testSuitesSourceSnapshotStorage.delete(key)
+        }.subscribe()
```

Comment:
why do you call subscribe here? maybe return `Mono<*>` and zip with Mono from method `deleteGit`?

Reply:
Changed